### PR TITLE
Mirroring GPG check

### PR DESCRIPTION
**Pull request by ikapelyukhin**
The PR is good to go for the review, however an ECO might be needed for it to be accepted by maintenance (due to added dependency for `gpg2` package in the spec file).

### CODE REVIEW

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/14435/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-run-tests. (Merged 82494e09d99195ebc80f1e4422e92534e4d89af0 into f06fa3c8d156af25e5bac0eb8afd66ef719d8921)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/14438/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-run-tests. (Merged 82494e09d99195ebc80f1e4422e92534e4d89af0 into f06fa3c8d156af25e5bac0eb8afd66ef719d8921)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/14440/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-run-tests. (Merged 8952faddb43d52983ce769548cdeb2b2eecab327 into f06fa3c8d156af25e5bac0eb8afd66ef719d8921)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/14441/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-run-tests. (Merged 8952faddb43d52983ce769548cdeb2b2eecab327 into f06fa3c8d156af25e5bac0eb8afd66ef719d8921)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/14442/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-run-tests. (Merged 8952faddb43d52983ce769548cdeb2b2eecab327 into f06fa3c8d156af25e5bac0eb8afd66ef719d8921)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/14444/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-run-tests. (Merged 8952faddb43d52983ce769548cdeb2b2eecab327 into f06fa3c8d156af25e5bac0eb8afd66ef719d8921)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/14446/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-run-tests. (Merged 8952faddb43d52983ce769548cdeb2b2eecab327 into f06fa3c8d156af25e5bac0eb8afd66ef719d8921)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/14448/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-run-tests. (Merged 8952faddb43d52983ce769548cdeb2b2eecab327 into f06fa3c8d156af25e5bac0eb8afd66ef719d8921)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/14684/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged a3733b57fdfbd84c91bb0d648e1cd264dcddb45e into f06fa3c8d156af25e5bac0eb8afd66ef719d8921)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/15037/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 40cd0758fb58716541f42000978b0fc96e44709e into f06fa3c8d156af25e5bac0eb8afd66ef719d8921)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/15051/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged f14b66e6ab305feb360470d55d119575c2f2a794 into f06fa3c8d156af25e5bac0eb8afd66ef719d8921)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/15250/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-build-images. (Merged d0b2378bdcf2efb392c1e33ab1b1ffc66e00dcde into f06fa3c8d156af25e5bac0eb8afd66ef719d8921)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/15258/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-build-images. (Merged d0b2378bdcf2efb392c1e33ab1b1ffc66e00dcde into f06fa3c8d156af25e5bac0eb8afd66ef719d8921)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/15260/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-build-images. (Merged d0b2378bdcf2efb392c1e33ab1b1ffc66e00dcde into f06fa3c8d156af25e5bac0eb8afd66ef719d8921)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/15265/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged a06519f0151e6706130a216c001666f9a5c162eb into f06fa3c8d156af25e5bac0eb8afd66ef719d8921)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/15330/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged a06519f0151e6706130a216c001666f9a5c162eb into 0f0459635cd7756df430ae881b79a8bdabe96a29)

File: Gemfile.lock [hunk 1]
```diff
@@ -126,7 +126,7 @@ GEM
     ffaker (2.11.0)
     ffi (1.9.25)
     formatador (0.2.5)
-    fuubar (2.4.0)
```
Comment (Sergeykot) [inline Gemfile.lock]: this is already in master, please rebase

File: lib/rmt/gpg.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
+require 'English'
+require 'fileutils'
+require 'tmpdir'
+
+class RMT::Gpg
```
Comment (Sergeykot) [inline lib/rmt/gpg.rb]: I think it should be `RMT::GPG` everywhere

File: lib/rmt/gpg.rb [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+    run_import_key
+    run_verify_signature
+
+    true
+  ensure
+    FileUtils.rm_rf(@tmpdir)
+  end
+
+  protected
```
Comment (Sergeykot) [inline lib/rmt/gpg.rb]: why `protected` and not `private` ?

**Comment by Sergeykot** [comment]
Except of the comments above - LGTM, please make requested changes and I will approve the PR

File: lib/rmt/gpg.rb (hunk 3 above)
Comment (ikapelyukhin) [reply lib/rmt/gpg.rb]: Why not? In Ruby there's very little difference between `protected` and `private`. I personally prefer `protected`, mostly due to how `protected` works in other languages.

File: Gemfile.lock (hunk 1 above)
Comment (ikapelyukhin) [reply Gemfile.lock]: I will, I fixed it before it was fixed in master to make sure that the tests can pass.

File: lib/rmt/gpg.rb (hunk 2 above)
Comment (ikapelyukhin) [reply lib/rmt/gpg.rb]: OK, fair enough.

File: Gemfile.lock (hunk 1 above)
Comment (ikapelyukhin) [reply Gemfile.lock]: Done.

File: lib/rmt/gpg.rb (hunk 2 above)
Comment (ikapelyukhin) [reply lib/rmt/gpg.rb]: Done.

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/16401/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged d6476d61d46ef5276dab671da05a5a4329ea3cff into 0f0459635cd7756df430ae881b79a8bdabe96a29)

File: lib/rmt/gpg.rb (hunk 3 above)
Comment (Sergeykot) [reply lib/rmt/gpg.rb]: exactly, so the thing is that we don't _need_ protected in this case. but now I check other places in the project where we also use `protected` so I guess it's OK :slightly_smiling_face:

**Pull request merged**